### DISCUSSION TITLE
Quadratic invitation rewards

### DISCUSSION THREAD

**Discussion by midenaio** [idena-network/idena-docs#101] [IIPs]
To stimulate the organic growth of the network we propose to change the invitation rewards distribution. 
Read more details [here](https://docs.idena.io/docs/iip/iip-6).

**Comment by 0JustinMiles0** [comment]
"If the inviter was penalized for a bad flip, the invitation rewards are not paid neither to the inviter nor to the invitee."
How will the invitation rewards be handled in case of the inviter being suspended or terminated/killed?

**Reply by Zen-44** [reply]
Should result in the same outcome since there won't be any reward to be shared with the inviter or invitee.

**Comment by 0JustinMiles0** [comment]
Quick thoughts on IIP-6:
- Invites won't be "fungible", it would be interesting to see how people could promote the value of their invite.
- It would certainly hurt flips farms that use the self-invite mechanism.
- Newly verified users with relatively small stake may delay inviting their network until they get more in stake to maximize rewards invite.
- More involved (=capitalized) users have more of an incentive to make an extra effort to invite people but they may already have exhausted their personal network and may be less incline to randomly promote invites to people they don't know.
- It will certainly help newcomers be more competitive for mining given they can obtain a high value invite. There is a risk that some legit newbie get further penalized for not knowing a well-capitalized participant which can be a debatable penalty.
- It strengthens the "web-of-trust" features of the Idena network as weakly capitalized accounts have less of an incentive to share invites.

Given that IIP-5 is successful at hurting pools and that the rewards are now shared with invitees, I have mostly a favorable opinion on this new IIP although I'd like to better understand what potential unintended consequences it could create. 

One suggestion I would make is to allow candidates to cancel the invitation they activate if they see that the inviter has lied on their stake amount or if they want to seek for a more valuable invite.

**Reply by 0JustinMiles0** [reply]
Second thoughts:

I'm not against the principle of IIP-6 but I think it needs two additional elements:
- An accessible mechanism for invitees to cancel their invite to disincentivize inviters from lying about their stake and allow invitees to seek more rewarding invites.
- A cliff and vesting mechanism to spread the payment of the reward to the inviter over a longer period of time to limit the abuse of self-invitation every three epochs. I would suggest a 3 epochs cliff and 10 validated epochs vesting schedule. Although IIP-6 doesn't create the mechanism for abusing self-invites, it makes it more attractive for identities with a  decent stake to exploit it. Indeed, since half of the rewards are now transferred to the invitee, there is an additional incentive to control the invitee's account. Also, the reward amount can be significantly higher and further motivate many stakers to take part in the self-invitation practice. A cliff and vesting mechanism would discourage this practice and at worst limit it to once every 13+ epochs. Also the payment of the rewards to the invitee should also be paid over a longer period of time. Ideally, this period should be at least half the time of the "cliff + vesting" period so that there is no incentive to terminate the invitee and send a new auto-invite for half of the invitation rewards every three epochs. This would also motivate the invitee to reach Human status and increase the confidence for the inviters that the invitees will maintain their identity until the vesting period has completed. An example of rewards payment schedule for an invitee could be over seven validated epochs: 5%, 5%, 10%, 10%, 20%, 20%, 30%.

**Comment by hlolve** [comment]
Stake for new IDs is not protected, so as fast as possible will kill the ID to get the stake, and we will see a circle of creating and killing IDs from high stake invitations. One ID with large stake, 1 or 2 invitations per epoch needs 5 workers to make this circle, or just a family pool and delegating to just 1 node.

**Reply by 0JustinMiles0** [reply]
I agree this is a problem. It seems the best strategy for high stakers would be to auto-invite their second identity then kill it to get the stake. In the meantime, low stakers with invites would have almost no incentive to even care about invites. Even if the demand to join the network is currently low, long term it seems that the risk would be that newcomers struggle finding invites. Quadratic invitation rewards would reduce the incentive for flips farms (unless they can compensate with a few high staker identities) to auto-invite but would increase it substantially for high stakers to a point that is probably not desirable.

**Reply by 0JustinMiles0** [reply]
Maybe a potential solution to the auto-invite abuse with IIP-6 would be to create an "invitation vault" in which 100% of the inviter rewards would be locked in. Those funds couldn't be transferred, wouldn't not count for staking rewards and would be burnt if the inviter identity is terminated. These invitation rewards would be progressively transferred to the main wallet following a vesting schedule. If the invitee is terminated, the remaining funds generated from this invitee would be burnt. The vesting period could be something like 10 epochs (from age 4 to age 14 of the invitee). The invitation rewards wouldn't start vesting until the invitee reaches age 4 (cliff period). The tradeoff is that these funds would be at risk during the entire vesting period which could be quite long and the total amount of invitation rewards that make it to the main wallet would be totally dependent on the invitee maintaining its identity during the entire vesting schedule. The benefit is that there would be no incentive to auto-invite a second identity as killing the second identity to collect the invitee rewards would burn the inviter rewards (zero-sum game).

**Comment by ubiubi18** [comment]
I support this IIP 100% .

**Comment by midenaio** [comment]
Thanks everyone for your feedback

@0JustinMiles0 @hlolve: Large-scale stakeholders seek to maximize the return on their capital. Since large-scale stakeholders are more discriminated against, they have a strong motivation to maintain the second identity with stake protection (with Human status) and distribute their iDNA evenly between the two identities.
Since even after IIP-6, mining returns will be higher than returns from successful invites, it is a rational strategy for large-scale stakeholders to use the second identity for mining, and not for abusing self-invites. The same applies to the third and fourth identities if it is a family pool.
